### DISCUSSION TITLE
Is the continuous batching function enabled by default in vllm?

### DISCUSSION THREAD

**Discussion by SeibertronSS** [vllm-project/vllm#547] [Q&A]
Is the continuous batching function enabled by default in vllm? Can this feature be turned on or off selectively?

**Comment by zhuohan123** [comment]
Yes, this is enabled by default and cannot be turned off. Turning off continuous batching requires a rewrite of our system architecture, which also brings no benefit in performance. Therefore, we did not implement this.

**Reply by Kelsey2018** [reply]
So, what the continuous batching specifically is in VLLM?

**Comment by zhaozhixin** [comment]
Does continues batching has a way to be adjusted? I found that the latency increased significantly when using api_server than offline inference.

**Comment by aniketmaurya** [comment]
How do we control the batch size and timeout for continous batching though?
